life_expectancy/cleaning.py

```diff
@@ -0,0 +1,188 @@
+    return None
+
+
+def split_combined_column(data: pd.DataFrame) -> pd.DataFrame:
+    """Splits the combined column into four separate columns: 'unit', 'sex', 'age', and 'region'."""
+    data_split_columns = data.iloc[:, COMBINED_COLUMN].str.split(',', expand=True)
+    data_split_columns.columns = ['unit', 'sex', 'age', 'region']
+   
+    # Drop the combined column from the original dataframe
+    data_reduced = data.drop(columns=[data.columns[0]])   
+    splitted_data = pd.concat([data_split_columns, data_reduced], axis=1)
+    
+    return splitted_data
+
+    
+def unpivot_dataframe(data: pd.DataFrame) -> pd.DataFrame:
+    """Unpivots the DataFrame to have 'year' and 'value' columns."""
+    unpivoted_data = data.melt(id_vars=['unit', 'sex', 'age', 'region'],
+                               var_name='year',
+                               value_name='value')
+    return unpivoted_data
+
+
+def transform_data(data: pd.DataFrame) -> pd.DataFrame:
+    """
+    Transforms the original DataFrame by splitting the combined column into four 
+    columns and unpivoting the DataFrame.
+    """
+    splitted_data = split_combined_column(data)
+    new_data = unpivot_dataframe(splitted_data)
+    
+    return new_data
+
+
+def remove_column_spaces(data: pd.DataFrame, column_name: str) -> pd.Series:
+    """Removes the spaces that might exist in the column's values."""
+    return data[column_name].str.strip()
+
+
+def filter_column_by_numeric_strings(data: pd.DataFrame, column_name: str) -> pd.DataFrame:
+    """Filters the DataFrame by the column values that are numeric."""
+    return data[data[column_name].str.isnumeric()]
+
+
+def change_column_type(data: pd.DataFrame, column_name: str, data_type: Choosable) -> pd.Series:
+    """Casts a given column to a given data type."""
+    return data[column_name].astype(data_type)
+
+
+def clean_year_column(data: pd.DataFrame) -> pd.DataFrame:
+    """Processes the 'year' column by performing a set of operations."""
+    data['year'] = remove_column_spaces(data, 'year')
+    filtered_data = filter_column_by_numeric_strings(data, 'year')
+    filtered_data['year'] = change_column_type(filtered_data, 'year', int)
+    
+    return filtered_data
+
+
+def remove_non_numeric_characters_from_column(data: pd.DataFrame, column_name: str) -> pd.Series:
+    """Removes all non-numeric characters from a given column."""
+    return data[column_name].apply(lambda x: re.sub(r'[^\d.]', '', str(x)))
+
+
+def convert_column_to_numeric(data: pd.DataFrame, column_name: str) -> pd.Series:
+    """Converts a given column to numeric and converts non-numerical values to NaN."""
+    return pd.to_numeric(data[column_name], errors='coerce')
+
+
+def remove_nan_by_column_subset(data: pd.DataFrame, column_subset: list[str]) -> pd.DataFrame:
+    """
+    Removes rows from a DataFrame based on the rows with NaN of a given subset 
+    of columns.
+    """
+    return data.dropna(subset=column_subset).reset_index(drop=True)
```

Comment:
Nit: Would not wrap this into a function, because the purpose of the pandas function is exactly that

Reply:
I thought I had responded to this comment, but it seems I may have forgotten to submit it.
The reason for implementing this function, as well as the previous one, in this way is the same reason I did so with the remove_non_numeric_characters_from_column() function: to make it reusable. Besides, since all three functions are called sequentially within the same function, it's important to maintain the same level of abstraction throughout.